spec/statement.md

```diff
@@ -0,0 +1,166 @@
+> [`current indentation level`]. In the indented syntax, no character other than
+> the `document indentation character` may be used for indentation.
+
+[`current indentation level`]: #current-indentation-level
+
+## Syntax
+
+### ScssStatements
+
+<x><pre>
+**ScssStatements**      ::= (Statement ';'?¹)* Statement?
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment`, `SilentComment`, or ends in a `Block`.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement [IndentSame])* Statement
+</pre></x>
+
+[IndentSame]: #indentation
+
+### Statements
+
+<x><pre>
+**Statements**          ::= ([ScssStatements] | [IndentedStatements])¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+1. Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements]
+**Block**          ::= (ScssBlock | IndentedBlock)¹
+</pre></x>
+
+[IndentMore]: #indentation
+
+1: Only the production for the current syntax is valid.
+
+### Comments
+
+#### LoudComment
+
+<x><pre>
+**ScssLoudComment**          ::= '/*' '!'? (.* | Interpolation)*'*/'
+**IndentedLoudComment**      ::= '/*' '!'? (.* | Interpolation)*
+**LoudComment**              ::= (ScssLoudComment | IndentedLoudComment)¹
+</pre></x>
+
+> TODO: Wildcard and character set regex syntax conflict with markdown.
+
+1: Only the production for the current syntax is valid.
+
+#### SilentComment
+
+<x><pre>
+**ScssSilentComment**          ::= '//' .*
+**IndentedSilentComment**      ::= '//' .*
+**SilentComment**              ::= (ScssSilentComment | IndentedSilentComment)¹
+</pre></x>
+
+1: Only the production for the current syntax is valid.
+
+#### WhitespaceComment
+
+<x><pre>
+**ScssWhitespaceComment**          ::= '//' .*'//' | '/*' .*'*/'
+**IndentedWhitespaceComment**      ::= '/*' .* '*/'
+**WhitespaceComment*¹              ::= ScssWhitespaceComment
+&#32;                                | IndentedWhitespaceComment
+</pre></x>
+
+1: Only the production for the current syntax is valid.
+
+### Whitespace
+
+<x><pre>
+**LineBreak**               ::= CarriageReturn | LineFeed | FormFeed
+**ScssWhitespace**          ::= LineBreak | Space | Tab
+**IndentedWhitespace**      ::= Space | Tab
+**Whitespace**              ::= (ScssWhitespace | IndentedWhitespace)¹
+&#32;                         | [WhitespaceComment]
+</pre></x>
+
+1: Only the production for the current syntax is valid.
+
+[WhitespaceComment]: #whitespacecomment
+
+### Indentation
+
+<x><pre>
+**WhitespaceOnlyLine**          ::= IndentSame (Whitespace)*
+**IndentCharacter**             ::= Space | Tab
+**IndentSame**                  ::= WhitespaceOnlyLine* [LineBreak]
```

Comment:
`WhitespaceOnlyLine*` should come after `LineBreak`; otherwise you're parsing trailing whitespace after the last statement as "a line".